src/idl_gen_go.cpp

```diff
@@ -858,110 +1111,269 @@ class GoGenerator : public BaseGenerator {
+        //  code += "\t// string \n";
+        code += "\t" + offset + ":= flatbuffers.UOffsetT(0)\n";
+        code += "\tif len(t." + MakeCamel(field.name) + ")> 0  {\n";
+        code += "\t\t" + offset + " = builder.CreateString(t." +
+                MakeCamel(field.name) + ")\n\t}\n";
+      }  // byte slice
+      else if (field.value.type.base_type == BASE_TYPE_VECTOR &&
+               field.value.type.element == BASE_TYPE_UCHAR &&
+               field.value.type.enum_def == nullptr) {
+        //    code += "\t// byte slice \n";
         code += "\t" + offset + " := flatbuffers.UOffsetT(0)\n";
         code += "\tif t." + MakeCamel(field.name) + " != nil {\n";
         code += "\t\t" + offset + " = builder.CreateByteString(t." +
                 MakeCamel(field.name) + ")\n";
         code += "\t}\n";
-      } else if (field.value.type.base_type == BASE_TYPE_VECTOR) {
-        code += "\t" + offset + " := flatbuffers.UOffsetT(0)\n";
-        code += "\tif t." + MakeCamel(field.name) + " != nil {\n";
+
+      }  // vector
+      else if (field.value.type.base_type == BASE_TYPE_VECTOR) {
+        // vector begin
         std::string length = MakeCamel(field.name, false) + "Length";
         std::string offsets = MakeCamel(field.name, false) + "Offsets";
-        code += "\t\t" + length + " := len(t." + MakeCamel(field.name) + ")\n";
+        // vector of string ( string array )
         if (field.value.type.element == BASE_TYPE_STRING) {
+          code += "\t" + offset + " := flatbuffers.UOffsetT(0)\n";
+          code += "\tif t." + MakeCamel(field.name) + " != nil {\n";
+
+          code += "\t\t" + offset + " = builder.StringsVector( t." +
+                  MakeCamel(field.name) + "...)\n";
+          code += "\t}\n";
+
+        }  //
+        else if (field.value.type.element == BASE_TYPE_UNION) {
+          code += "\t// vector of unions \n";
+
+          std::string offset_type = MakeCamel(field.name, false) + "TypeOffset";
+
+          code += "\t" + offset + " := flatbuffers.UOffsetT(0)\n";
+          code += "\t" + offset_type + " := flatbuffers.UOffsetT(0)\n";
+
+          code += "\tif t." + MakeCamel(field.name) + " != nil {\n";
+          code +=
+              "\t\t" + length + " := len(t." + MakeCamel(field.name) + ")\n";
+
           code += "\t\t" + offsets + " := make([]flatbuffers.UOffsetT, " +
                   length + ")\n";
-          code += "\t\tfor j := 0; j < " + length + "; j++ {\n";
-          code += "\t\t\t" + offsets + "[j] = builder.CreateString(t." +
-                  MakeCamel(field.name) + "[j])\n";
+
+          code += "\t\tfor j := " + length + " - 1; j >= 0; j-- {\n";
+          code += "\t\t\t" + offsets + "[j] = t." + MakeCamel(field.name) +
+                  "[j].Pack(builder)\n";
           code += "\t\t}\n";
-        } else if (field.value.type.element == BASE_TYPE_STRUCT &&
-                   !field.value.type.struct_def->fixed) {
+
+          code += "\t\t" + struct_def.name + "Start";
```

Comment:
This is a lot of code and I don't really understand it. Could you add thorough comments to this C++ code?

Reply:
ok.  i should add comments in  src/idl_gen_go.cpp.

Reply:
@rw 
i had split long code segments, added comments, fix test code. 
thanks for review ...